pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -466,6 +498,12 @@ func (c *Reconciler) createPod(tr *v1alpha1.TaskRun, ts *v1alpha1.TaskSpec, task
 		return nil, err
 	}
 
+	err = resources.AddOutputImageDigestExporter(tr, ts, c.resourceLister, c.Logger)
```

Comment:
do you think it could make sense for `resource.AddOutputResources` to take care of adding _all_ output related images, including the image digest exporter (which is how the storage outputs seem to work)?

Reply:
its because this wasn't really getting the container spec from the resource, its a special case where we added one container at the end, Jason suggested to add this after every step, so I'll look into refactoring it anyways